### ISSUE TITLE
HTTP Request node: wrong http exception handling

### ISSUE THREAD

**Issue by faxioman** [node-red/node-red#4150]
### Current Behavior

When you enable the option "Only send non-2xx responses to Catch node" in an http request node if you get an http exception from server, the node alway sends output and the catch node never gets the exception.

### Expected Behavior

Enabling "Only send non-2xx responses to Catch node" should send message only to catch node

### Steps To Reproduce

Try an http request node with "Only send non-2xx responses to Catch node"

### Example flow

```
[
    {
        "id": "c72880672de95bf5",
        "type": "http request",
        "z": "aef408433c5496a0",
        "name": "",
        "method": "POST",
        "ret": "txt",
        "paytoqs": "ignore",
        "url": "https://www.google.it/",
        "tls": "",
        "persist": true,
        "proxy": "",
        "insecureHTTPParser": false,
        "authType": "",
        "senderr": true,
        "headers": [],
        "x": 270,
        "y": 1160,
        "wires": [
            [
                "1537d57cf3ade772"
            ]
        ]
    },
    {
        "id": "1537d57cf3ade772",
        "type": "debug",
        "z": "aef408433c5496a0",
        "name": "debug 9",
        "active": true,
        "tosidebar": true,
        "console": false,
        "tostatus": false,
        "complete": "true",
        "targetType": "full",
        "statusVal": "",
        "statusType": "auto",
        "x": 500,
        "y": 1160,
        "wires": []
    },
    {
        "id": "a3b7bf20f113b9de",
        "type": "inject",
        "z": "aef408433c5496a0",
        "name": "",
        "props": [
            {
                "p": "payload"
            },
            {
                "p": "topic",
                "vt": "str"
            }
        ],
        "repeat": "",
        "crontab": "",
        "once": false,
        "onceDelay": 0.1,
        "topic": "",
        "payload": "",
        "payloadType": "date",
        "x": 100,
        "y": 1160,
        "wires": [
            [
                "c72880672de95bf5"
            ]
        ]
    },
    {
        "id": "84d5d28b82cfe4aa",
        "type": "catch",
        "z": "aef408433c5496a0",
        "name": "",
        "scope": [
            "c72880672de95bf5"
        ],
        "uncaught": false,
        "x": 270,
        "y": 1220,
        "wires": [
            [
                "6f440eab87b1b68c"
            ]
        ]
    },
    {
        "id": "6f440eab87b1b68c",
        "type": "debug",
        "z": "aef408433c5496a0",
        "name": "debug 10",
        "active": true,
        "tosidebar": true,
        "console": false,
        "tostatus": false,
        "complete": "true",
        "targetType": "full",
        "statusVal": "",
        "statusType": "auto",
        "x": 500,
        "y": 1220,
        "wires": []
    }
]
```

### Environment

- Node-RED version: 3.0.2
- Node.js version: 16.17.1
- npm version: 8.15.0
- Platform/OS: MacOS / Ubuntu Linux
- Browser: Safari/Chrome/Edge

**Comment by faxioman** [comment]
Also check this issue: https://github.com/node-red/node-red/issues/3082
I see the reason behind setting:
```
opts.throwHttpErrors = false;
```
Anyway, we need to change the way exceptions are managed.

**Comment by knolleary** [comment]
This is the relevant comment in the code:

https://github.com/node-red/node-red/blob/32a49a1ef17894b6dd2dd83b5b2450f94ab3a27f/packages/node_modules/%40node-red/nodes/core/network/21-httprequest.js#L629-L633

We didn't make the suggested change in 3.x.

We do need to more clearly define the desired behaviour. I propose the following:

1. Enable the 'Only send non-2xx responses to Catch node` by default for *new* HTTP Request nodes (existing nodes keep their same behaviour)
2. If that option is enabled
   - any 2xx response will be emitted by the node in the normal manner
   - anything else (including network level errors) will be logged as an error for a Catch node to handle

That keeps the golden path on the main flow, and allows all errors to be handled via a Catch node. I think that is the most expected behaviour.

As it will be a change in behaviour, we'll have to save it for 4.x - which will be coming fairly soon once 3.1 is out of the way.

**Comment by Shocktrooper** [comment]
Hello! looking for any updates on this as I just ran into this and confused why a catch node was not working at all

**Comment by skylord123** [comment]
Yeah I am pretty sure this is not working how it is supposed to based on the description. When I have `Only send non-2xx responses to Catch node` enabled I would expect any request that has a response code that is not 2xx to be sent to the catch node. This does not happen. It doesn't send to the catch node at all.

Based on the description this is not working as intended. If it was instead labeled "Only send exceptions to Catch node" that would be correct.

Here is a flow that shows this is not working as intended: https://gist.github.com/skylord123/602f67af7946b6cfa9492b33c01f808e

I have some flows that call a ton of http request nodes one after the next and want it to just stop the flow and output to the catch node if there is an error. Now I have to handle that with a switch node after every http request node and it gets out of hand quick.

For now I am just going to clone the http request node into my own module to fix this. I was really hoping to avoid that but there doesn't seem to be a good alternative.
